app/services/account_service.rb

```diff
@@ -2,18 +2,42 @@
 
 # Retrieve names from the account API run by UIT
 class AccountService
+  class AccountServiceHiccough < StandardError; end
+
   def initialize
     pem_file = File.read(Settings.accountws.pem_file)
     @key = OpenSSL::PKey.read pem_file
     @cert = OpenSSL::X509::Certificate.new pem_file
   end
 
   def fetch(sunetid)
-    Rails.cache.fetch(sunetid, namespace: 'account', expires_in: 1.month) do
+    # Never write a `nil` to the cache or legit users will have bogus cache
+    # entries for the duration of the expiry period (one month)
+    Rails.cache.fetch(sunetid, namespace: 'account', expires_in: 1.month, skip_nil: true) do
       url = template.partial_expand(sunetid:).pattern
-      response = connection.get(url)
-      doc = response.body
-      doc.slice('name', 'description')
+
+      # The account service frequently returns 500 errors. Retry the connection five times in rapid succession.
+      begin
+        tries ||= 1
+        response = connection.get(url).body
+        # Raise and retry if the response is an HTTP 500.
+        #
+        # If, on the other hand, a bogus sunetid is provided, the `status` of the response will be 404, and then we:
+        #
+        # 1. Do *not* want to retry; but
+        # 2. *Do* want to cache the empty document
+        raise AccountServiceHiccough if response['status'] == 500
```

Comment:
small niggle about how `response` is the body of the response (maybe `response_body` is clearer?)  - this also count on the putting the status in the body of the response as opposed to checking the actual http response code itself (presuming they are doing the correct thing and actually returning a 500 code when there is an error)

Reply:
@peetucket Good call. I renamed the variable to `response_body`. They do include the HTTP status in the body for both 404 and 500 errors, at least.